### ISSUE TITLE
Custom Development Client Error When Compiled Against Android SDK 33

### ISSUE THREAD

**Issue by mattThousand** [expo/expo#18658]
### Summary

If `compileSDKVersion` and `targetSDKVersion` are set to `33` in `app/build.gradle`, any subsequent custom development client generated will fail to run on Android with the error:

```
Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
          This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
```

### What platform(s) does this occur on?

Android

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.5
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
      Watchman: 2022.02.21.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /Users/matthewbuckley/gems/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8815526
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~46.0.7 => 46.0.7
      react: 18.0.0 => 18.0.0
      react-dom: 18.0.0 => 18.0.0
      react-native: 0.69.4 => 0.69.4
      react-native-web: ~0.18.7 => 0.18.7
    npmGlobalPackages:
      eas-cli: 0.60.0
      expo-cli: 6.0.2
    Expo Workflow: bare
```

### Minimal reproducible example

- `npx create-expo-app my-app`
- `cd my-app`
- `expo install expo-dev-client`
- `expo prebuild`
- Set `compileSDKVersion` and `targetSDKVersion` to `33` in `app/build.gradle`
- Generate a custom development client (internal distribution) build using `eas build`
- Install the build on a physical device
- run `expo start --dev client`
- run `a` to open on android
- Error will show

**Comment by lukmccall** [comment]
Thanks for the report and the repro, @mattThousand. I was able to reproduce this bug. We'll be working on a fix soon.

**Comment by Yonom** [comment]
Compiling our rn project with expo-dev-client installed and `compileSdkVersion = 33` but `targetSdkVersion 31` causes a different issue in `expo-dev-menu`:

```
> Task :expo-dev-menu:compileDebugKotlin FAILED
e: node_modules/expo-dev-menu/vendored/react-native-gesture-handler/android/devmenu/com/swmansion/gesturehandler/PanGestureHandler.kt: (57, 36): 
Type mismatch: inferred type is Context? but Context was expected
```

I will create a fix for this in a minute...

**Comment by AdamDiament** [comment]
I raised [this issue ](https://github.com/OneSignal/onesignal-expo-plugin/issues/107)on the expo plugin Github, however I think it is related to expo itself rather than onesignal 

TLDR - I have no Bluetooth in my app but I get this error in logcat when setting compileSDKVersion and targetSDKVersion to 33 with the build properties plugin


```
08-28 13:40:08.570 19612 19697 E ReactNativeJS: Error: Exception in HostObject::get(propName:NativeUnimoduleProxy): java.lang.SecurityException: Settings key: <bluetooth_name> is only readable to apps with targetSdkVersion lower than or equal to: 31
08-28 13:40:08.575 19612 19697 E ReactNativeJS: Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
08-28 13:40:08.575 19612 19697 E ReactNativeJS:       This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
```

**Comment by ahnafnafee** [comment]
Still encountering this issue in Pixel 7 with Android 13 
```js
Error: Exception in HostObject::get for prop 'NativeUnimoduleProxy': java.lang.SecurityException: Settings key: <bluetooth_name> is only readable to apps with targetSdkVersion lower than or equal to: 31
```

Found a similar issue here: https://github.com/expo/expo/issues/18738 with similar test devices (Android 13)

**Comment by Car-ElWilliams** [comment]
Still an issue
